_drafts/fwup-signing.md

```diff
@@ -0,0 +1,639 @@
+---
+title: "Authenticate firmware builds with code signing"
+description: WIP
+author: francois
+---
+
+<!-- excerpt start -->
+<!-- excerpt end -->
+
+{% include newsletter.html %}
+{% include toc.html %}
+
+## Firmware Signing Explained
+
+### What is code signing
+
+Code signing is a method of proving a file was created by a trusted source and
+has not been tampered with. This is achieved by creating a signature for the
+file: a token which can be verified but not forged.
+
+We all use signatures in our day to day life, by scribbling our name at the
+bottom of our credit card receipts, contracts, and bank checks. These signatures
+can be verified quite easily by comparing them to previous signatures, but are
+hard to forge because they rely on an individual's muscle memory.
+
+Code signing uses cryptographic algorithms to achieve a similar goal. The
+resulting signatures can only be generated by people who know a given password,
+also known as a secret key, but can be verified as genuine by anyone.
+
+Additionally, the signatures are tied to a given firmware binary so should the
+binary change by so much as 1 bit the signature would no longer be valid.
+
+### Why sign our firmware
+
+By implementing signature verification in our bootloader we can identify
+whether or not a given firmware update was provided by the manufacturer, or if
+it has been tampered with. The bootloader can then decide to either warn the
+user, void the device's warranty, or simply refuse to run the unauthenticated
+binary.
+
+With more and more devices connected to the internet, security is an
+increasingly hot topic in firmware development. A device which accepts firmware
+updates over the wireless or internet connectivity but does not verify it opens
+itself to compromise. By feeding it with a malicious firmware image, an attacker
+might:
+* Brick the device, or the whole fleet
+* Snoop on end users and compromise their privacy and security
+* Strategically malfunction at a critical time
+
+These are highly undesirable outcomes, which can be effected at scale due to the
+internet of things. In 2020, it is reckless to implement firmware update for our
+systems without some form of authentication.
+
+> **What signing is not**: code signing is an important component of firmware
+> security, it is not by itself sufficient to build secure systems. Secure
+> coding, static analysis, hardware tamper detection, JTAG locking, and many
+> more techniques should be implemented as well.
+> Code signing also does nothing to protect against reverse engineering. It is a
+> distinct technique from firmware encryption.
+
+### ECDSA
```

Comment:
I think this section is quite dry, and less exciting than FIrmware Signing implementation. 

Maybe you can quickly say `We use ECDSA, which we talk about [later in the post](#ecdsa)` and jump straight to the example, and place this section later.

Reply:
I added a "jump to implementation" at the top of the theory section. At the same time, I think this needs to be before the implementation, because it details the use of SHA-256 which is key to the implementation.